src/simulation.cpp

```diff
@@ -429,18 +445,29 @@ void finalize_generation()
       }
     }
 
-  } else if (settings::run_mode == RunMode::FIXED_SOURCE) {
-    // For fixed-source mode, we need to sample the external source
-    simulation::time_sample_source.start();
-    fill_source_bank_fixedsource();
-    simulation::time_sample_source.stop();
   }
 }
 
 void initialize_history(Particle* p, int64_t index_source)
 {
   // set defaults
-  p->from_source(&simulation::source_bank[index_source - 1]);
+  if(settings::run_mode == RunMode::FIXED_SOURCE) {
+	// initialize random number seed
+	int64_t id = (simulation::total_gen + overall_generation() - 1)*settings::n_particles +
+	  simulation::work_index[mpi::rank] + index_source;
+	uint64_t seed = init_seed(id, STREAM_SOURCE);
+	// sample from external source distribution or custom library then set
+	Particle::Bank site;
+	if (!settings::path_source_library.empty()) {
+	  site = sample_custom_source_library(&seed);
+	} else {
+	  site = sample_external_source(&seed);
+	}
+	p->from_source(&site);
+  } else if (settings::run_mode == RunMode::EIGENVALUE) {
+	// set defaults for eigenvalue simulations from primary bank
+    p->from_source(&simulation::source_bank[index_source - 1]);
+  }
```

Comment:
More TABs messing up indentation: 

```suggestion
  if(settings::run_mode == RunMode::FIXED_SOURCE) {
    // initialize random number seed
    int64_t id = (simulation::total_gen + overall_generation() - 1)*settings::n_particles +
      simulation::work_index[mpi::rank] + index_source;
    uint64_t seed = init_seed(id, STREAM_SOURCE);
    // sample from external source distribution or custom library then set
    Particle::Bank site;
    if (!settings::path_source_library.empty()) {
      site = sample_custom_source_library(&seed);
    } else {
      site = sample_external_source(&seed);
    }
    p->from_source(&site);
  } else if (settings::run_mode == RunMode::EIGENVALUE) {
    // set defaults for eigenvalue simulations from primary bank
    p->from_source(&simulation::source_bank[index_source - 1]);
  }
```

Reply:
Done